app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -246,11 +249,41 @@ private void lookupUrlAndDownload(String url) {
                 .observeOn(Schedulers.io())
                 .subscribe(this::startFeedDownload,
                         error -> {
-                            showNoPodcastFoundError();
-                            Log.e(TAG, Log.getStackTraceString(error));
+                            if (error instanceof FeedUrlNotFoundException) {
+                                tryToRetrieveFeedUrlBySearch((FeedUrlNotFoundException) error);
+                            } else {
+                                showNoPodcastFoundError();
+                                Log.e(TAG, Log.getStackTraceString(error));
+                            }
                         });
     }
 
+    private void tryToRetrieveFeedUrlBySearch(FeedUrlNotFoundException error) {
```

Comment:
I would maybe show a `Snackbar` to the user which explains what happens. Otherwise, we might run into a situation where clicking one podcast opens the feed of another. That snackbar at least gives users a hint. Something like:

> The search result did not have a link to a podcast RSS address but AntennaPod tried to look up the podcast using another search engine.

Reply:
Should the snackbar be displayed if no matching podcast is found?

Depending on the response time of the search and if no matching podcast is found, the user may not have time to read the snackbar or may receive the error "No podcast was provided that can be displayed" and the snackbar at the same time.

Maybe i can display the snackbar "The suggested podcast did not have an RSS link, AntennaPod found a podcast that could match" only if a matching podcast is found.

Reply:
> Maybe i can display the snackbar "The suggested podcast did not have an RSS link, AntennaPod found a podcast that could match" only if a matching podcast is found.

Sounds good 👍